python/cuml/dask/common/comms.py

```diff
@@ -13,175 +13,72 @@
 # limitations under the License.
 #
 
-from cuml.nccl import nccl
-from cuml.dask.common.ucx import UCX
+import time
+from collections import OrderedDict
+import warnings
 
-import weakref
+from dask.distributed import get_worker
 
 from .comms_utils import inject_comms_on_handle, \
-    inject_comms_on_handle_coll_only, is_ucx_enabled
-from .utils import parse_host_port
+    inject_comms_on_handle_coll_only
 from cuml.common.handle import Handle
-import cuml.common.logger as logger
-
-from dask.distributed import get_worker, default_client
-
-from cuml.common.import_utils import has_ucp
-import warnings
-
-import time
-import uuid
 
-
-_global_comms = weakref.WeakValueDictionary()
-_global_comms_index = [0]
-
-
-def _set_global_comms(c):
-    if c is not None:
-        _global_comms[_global_comms_index[0]] = c
-        _global_comms_index[0] += 1
-
-
-def _del_global_comms(c):
-    for k in list(_global_comms):
-        try:
-            if _global_comms[k] is c:
-                del _global_comms[k]
-        except KeyError:
-            pass
-
-
-def worker_state(sessionId=None):
-    """
-    Retrieves cuML comms state on local worker for the given
-    sessionId, creating a new session if it does not exist.
-    If no session id is given, returns the state dict for all
-    sessions.
-    :param sessionId:
-    :return:
-    """
-    worker = get_worker()
-    if not hasattr(worker, "_cuml_comm_state"):
-        worker._cuml_comm_state = {}
-    if sessionId is not None and sessionId not in worker._cuml_comm_state:
-        # Build state for new session and mark session creation time
-        worker._cuml_comm_state[sessionId] = {"ts": time.time()}
-
-    if sessionId is not None:
-        return worker._cuml_comm_state[sessionId]
-    return worker._cuml_comm_state
-
-
-def get_ucx():
-    """
-    A simple convenience wrapper to make sure UCP listener and
-    endpoints are only ever assigned once per worker.
-    """
-    if "ucx" not in worker_state("ucp"):
-        worker_state("ucp")["ucx"] = UCX.get()
-    return worker_state("ucp")["ucx"]
-
-
-def _get_global_comms():
-    L = sorted(list(_global_comms), reverse=True)
-    for k in L:
-        c = _global_comms[k]
-        if c.nccl_initialized and (not c.comms_p2p or c.ucx_initialized):
-            return c
-        else:
-            del _global_comms[k]
-    del L
-    return None
-
-
-def default_comms(comms_p2p=False, client=None):
-    """ Return a comms instance if one has been initialized.
-        Otherwise, initialize a new comms instance.
-    """
-    c = _get_global_comms()
-    if c:
-        return c
-    else:
-        cb = CommsContext(comms_p2p, client)
-        cb.init()
-
-        _set_global_comms(cb)
-
-        return _get_global_comms()
-
-
-def _func_ucp_listener_port():
-    return get_ucx().listener_port()
```

Comment:
This whole file should be able to go away. The algorithms should just be able to use the RAFT comms directly.